persistent/Database/Persist/TH.hs

```diff
@@ -2646,16 +2647,17 @@ mkJSON mps (fixEntityDef -> def) = do
     requireExtensions [[FlexibleInstances]]
     pureE <- [|pure|]
     apE' <- [|(<*>)|]
+
+    let objectE = VarE 'object
+        withObjectE = VarE 'withObject
+        dotEqualE = VarE '(.=)
+        dotColonE = VarE '(.:)
+        dotColonQE = VarE '(.:?)
 #if MIN_VERSION_aeson(2,0,0)
-    toKeyE <- [|Key.fromString|]
+        toKeyE = VarE 'Key.fromString
 #else
-    toKeyE <- [|pack|]
+        toKeyE = VarE 'pack
 #endif
-    dotEqualE <- [|(.=)|]
-    dotColonE <- [|(.:)|]
-    dotColonQE <- [|(.:?)|]
-    objectE <- [|object|]
-    withObjectE <- [|withObject|]
```

Comment:
Bug was here. `withObject` was not imported, so I think this turned into an `UnboundVarE` instead of a `VarE` that we want. Wow. Using the `VarE 'withObject` form forces the name to be in scope, which then becomes a `NameG` value.

@danbroooks TIL 😅